go/oasis-node/cmd/common/metrics/resource.go

```diff
@@ -0,0 +1,62 @@
+// Package metrics implements a prometheus metrics service.
+package metrics
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/service"
+)
+
+// ResourceMicroService is interface for monitoring resources (cpu, mem, disk, net).
+type ResourceMicroService interface {
```

Comment:
Maybe `ResourceCollector interface` would be nicer, but no strong opinions about it.